lib/src/revset.rs

```diff
@@ -1977,20 +2015,29 @@ impl RevsetExtensions {
 pub struct RevsetParseContext<'a> {
     aliases_map: &'a RevsetAliasesMap,
     user_email: String,
+    /// The current local time when the revset expression was written.
+    now: NaiveDateTime,
+    /// The offset from UTC at the time the revset expression was written. TODO:
+    /// It would be better if this was the TimeZone so times could be computed
+    /// correctly across DST shifts.
+    offset: FixedOffset,
     extensions: &'a RevsetExtensions,
     workspace: Option<RevsetWorkspaceContext<'a>>,
 }
 
 impl<'a> RevsetParseContext<'a> {
-    pub fn new(
+    pub fn new<Tz: TimeZone>(
         aliases_map: &'a RevsetAliasesMap,
         user_email: String,
+        now: DateTime<Tz>,
         extensions: &'a RevsetExtensions,
         workspace: Option<RevsetWorkspaceContext<'a>>,
     ) -> Self {
         Self {
             aliases_map,
             user_email,
+            now: now.naive_local(),
+            offset: now.offset().fix(),
```

Comment:
I don't think `Tz` parameter makes sense here. If we convert `now` to `FixedOffset`, we can use `backend::Timestamp` type for API consistency.

If we want to support DST thingy, `DateTime<Local>` shouldn't be decomposed here?

Reply:
backend::Timestamp didn't seem appropriate because I eventually need the `DateTime<Tz>` to pass into `chrono_english::parse_date_string`. I didn't want to lose the time zone information. I resolved this by:

- Creating a DatePatternContext, which is either `Local(DateTime<Local>)` or `Utc(DateTime<Utc>)`
- Storing a DatePatternContext in the RevsetParseContext instead of any chrono types
- Call `context.to_date_pattern(s, kind)` to create the DatePattern

The production code uses `Local(DateTime<Local>)`, but tests use `Utc(DateTime<Local>)`.